nestkernel/parameter.cpp

```diff
@@ -273,4 +275,92 @@ RedrawParameter::value( RngPtr rng,
   return value;
 }
 
+
+ExpDistParameter::ExpDistParameter( const DictionaryDatum& d )
+  : p_( getValue< ParameterDatum >( d, "x" )->clone() )
+  , beta_( getValue< double >( d, "beta" ) )
+{
+  parameter_is_spatial_ = true;
+}
+
+double
+ExpDistParameter::value( RngPtr rng,
+  const std::vector< double >& source_pos,
+  const std::vector< double >& target_pos,
+  const AbstractLayer& layer )
+{
+  return std::exp( -p_->value( rng, source_pos, target_pos, layer ) / beta_ );
+}
+
+GaussianParameter::GaussianParameter( const DictionaryDatum& d )
+  : p_( getValue< ParameterDatum >( d, "x" )->clone() )
+  , mean_( getValue< double >( d, "mean" ) )
+  , std_( getValue< double >( d, "std" ) )
+  , two_std2_( 2 * std_ * std_ )
+{
+  parameter_is_spatial_ = true;
+}
+
+double
+GaussianParameter::value( RngPtr rng,
+  const std::vector< double >& source_pos,
+  const std::vector< double >& target_pos,
+  const AbstractLayer& layer )
+{
+  const auto x = p_->value( rng, source_pos, target_pos, layer );
+  return std::exp( -( ( x - mean_ ) * ( x - mean_ ) ) / two_std2_ );
+}
+
+
+Gaussian2DParameter::Gaussian2DParameter( const DictionaryDatum& d )
+  : px_( getValue< ParameterDatum >( d, "x" )->clone() )
+  , py_( getValue< ParameterDatum >( d, "y" )->clone() )
+  , mean_x_( getValue< double >( d, "mean_x" ) )
+  , mean_y_( getValue< double >( d, "mean_y" ) )
+  , std_x_( getValue< double >( d, "std_x" ) )
+  , std_y_( getValue< double >( d, "std_y" ) )
+  , rho_( getValue< double >( d, "rho" ) )
+  , std_x2_( std_x_ * std_x_ )
+  , std_y2_( std_y_ * std_y_ )
+  , std_xy_( std_x_ * std_y_ )
+  , two_sub_2rho2_( 2. * ( 1. - rho_ * rho_ ) )
```

Comment:
I do not think I understand what the first `2` (not the `two` at the beginning) in the name comes from. The last is the `rho_` squared, but what is the first one? Shouldn't it be `two_sub_rho2` (or `two_one_sub_rho2`, but that is a bit too much..)

Reply:
The idea for the name was from the expanded expression: `2*(1 - rho_*rho_) = 2 - 2*rho_**2`. But I've changed the naming scheme now.